packages/product-components/src/components/SelectAssetModal/SelectAssetModal.tsx

```diff
@@ -1,145 +1,201 @@
 
-export type SelectAssetModalProps = {
-    onSelectAssetModal: (selectedAsset: string) => void;
-    onFavoriteClick: (isFavorite: boolean) => void;
-};
-
-const tokens = ['ethereum', 'bitcoin', 'tether'];
+const getData = (options: SelectAssetOptionProps[]): AssetProps[] =>
+    options
+        .filter(item => item.type === 'currency')
+        .map(item => ({
+            cryptoId: item.value,
+            symbol: item.label,
+            name: item.cryptoName ?? item.label,
+            badge: item.networkName,
+            coingeckoId: item.coingeckoId,
+            contractAddress: item.contractAddress,
+            isFavorite: false,
+            height: ITEM_HEIGHT,
+        }));
 
 export const SelectAssetModal = ({
+    options,
+    networkCategories,
     onSelectAssetModal,
     onFavoriteClick,
+    onClose,
 }: SelectAssetModalProps) => {
+    const intl = useIntl();
     const [search, setSearch] = useState('');
-    const [end, setEnd] = useState(10000);
-    const [data, setData] = useState(getData(end, tokens)); // @TODO
+    const [searchCategory, setSearchCategory] = useState<SelectAssetSearchCategoryType>(null); // coingeckoNativeId as fallback for ex. polygon
+    const [end, setEnd] = useState(options.length);
+    const data = useMemo(() => getData(options), [options]);
     const { scrollElementRef, onScroll, ShadowTop, ShadowBottom, ShadowContainer } =
         useScrollShadow();
+    const networkCount = options.filter(item => item.type === 'group' && item.networkName).length;
 
-    useEffect(() => {
-        // simulate async data loading
-        setTimeout(() => {
-            setData(getData(end, tokens));
-        }, 100);
-    }, [end]);
-
-    const filteredData = data.filter(
-        item =>
-            item.symbol.includes(search) ||
-            item.name.includes(search) ||
-            item.badge.includes(search),
-    );
+    const filteredData = data.filter(item => {
+        const categoryFilter = searchCategory
+            ? item.coingeckoId === searchCategory.coingeckoId ||
+              item.coingeckoId === searchCategory.coingeckoNativeId
+            : true;
+        const searchFor = (property: string | undefined) =>
+            property?.toLocaleLowerCase().includes(search.toLocaleLowerCase());
 
-    const { parentElevation } = useElevation();
+        return (
+            (searchFor(item.name) ||
+                searchFor(item.cryptoId) ||
+                searchFor(item.badge) ||
+                searchFor(item.symbol)) &&
+            categoryFilter
+        );
+    });
 
     const shadowColor = mapElevationToBackgroundToken({
-        $elevation: parentElevation,
+        $elevation: 0,
     });
 
     return (
-        <NewModal heading="Select a token" onCancel={() => null} size="small">
+        <NewModal
+            heading={intl.formatMessage({
+                id: 'TR_SELECT_TOKEN',
+                defaultMessage: 'Select a token',
+            })}
```

Comment:
Should not we use `translationString` from `useTranslation` instead of `intl.formatMessage`? (I see two ways - which is correct?)